cli/plasmo/src/features/env/env-config.ts

```diff
@@ -50,22 +50,17 @@ function cascadeEnv(loadedEnvFiles: LoadedEnvFiles) {
   for (const { contents, name } of loadedEnvFiles) {
     try {
       envFileSet.add(name)
-      const result = dotenvExpand({
```

Comment:
I found no real benefit to the side-effects introduced by `dotenvExpand`, since there is explicit `process.env` var assignment in the iteration of the values from the `.env` files. If I'm mistaken, let me know, but leaving it in also conflicts with the ability to introduce this feature since it implicitly hydrates `process.env`

Reply:
(Gathering my thought) the main idea with dotenvExpand is to have this:

```
PASSWORD="s1mpl3"
DB_PASS="prefix-$PASSWORD"
```

And the reason we want that is to have the same behavior as nextjs and other frameworks with built-in env parsing.

> but leaving it in also conflicts with the ability to introduce this feature since it implicitly hydrates `process.env`

Can we run the expansion at the end of this loop? (on the parsed object)?

Perhaps that might be best. I.e on line 72, we'd do:

```
return dotenvExpand.expand({ parsed })
```

That way, we can hydrate the key for the JSON value manually first, then expand once we know everything's in-place?

Reply:
Ah ok, that explanation makes more sense. I'll play around with it a bit more and make sure that still works

Reply:
Fixed in 47059a18cfb247726126d389b1367654ba01ae70

Reply:
Ok, I think my latest commit resolves this, but maybe you have some specific use cases you can test?

Reply:
Superb - works for me! Tested on some variant with the dnr as well!